packages/Confirmation/src/Confirmation.js

```diff
@@ -73,29 +74,27 @@ const Confirmation = props => {
     popoverKey = uuid();
   }, [isPending]);
 
-  const triggerProps = {
-    isConfirmOpen,
-    handleOpenConfirm,
-  };
-
   const handleCloseConfirm = () => {
-    if (isConfirmOpen) {
-      setIsConfirmOpen(false);
-      if (triggerRef.current) triggerRef.current.focus();
-      setTimeout(onClose, 250);
-    }
+    setIsConfirmOpen(prevIsConfirmOpen => {
+      if (prevIsConfirmOpen) {
+        if (triggerRef.current) triggerRef.current.focus();
+        setTimeout(onClose, 250);
+      }
+      return false;
```

Comment:
why does it need to return `false`? If you leave this out it will implicitly return `undefined`, won't it? Which should be fine?

Reply:
isConfirmOpen should be boolean value, just by it's name, why not set it as a boolean value? Seems more logical to me

Reply:
Oh I see now, I didn't read it well before with the inline comments on.  

But I also now notice that there are side effects in your `setState` which I'm not so sure is a good practice.  (I'm not _sure_ it's a bad practice, it just makes my spidey sense tingle).

The `prevIsConfirmOpen` in the callback for `setState` is the same as `isConfirmOpen` before you call it though, right? So you _could_ just do that block first, then call `setState`?

Reply:
You mean like we were doing before?
```
if (isConfirmOpen) {	    
      setIsConfirmOpen(false);
}
```
The reason I changed it to do it this was was because the `isConfirmOpen` value in state was being evaluated as `true` twice as this function is called on button click and popover close event. even though it has been set to false. This approach ensures there are no async problems.